ql/src/java/org/apache/hadoop/hive/ql/udf/TimestampCastRestrictorResolver.java

```diff
@@ -45,7 +45,7 @@
   public TimestampCastRestrictorResolver(UDFMethodResolver parentResolver) {
     this.parentResolver = parentResolver;
     SessionState ss = SessionState.get();
-    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION)) {
+    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_CHECKS_TYPE_SAFETY)) {
```

Comment:
If I understand well this class is used to restrict casting timestamp/date to boolean, double, byte, float, integer, long, short values. I am not sure why we should deal with these checks at this point but I if we really need this then I guess it makes sense to extend it so that we apply the same checks for all types under `hive.strict.checks.type.safety` property. Should we create another JIRA for this?

Reply:
Hey @zabetak -- had a chat with @kgyrtkirk  regarding this.
Seems like udfs do not communicate their accepted format/s as they may introduce converters/Bridges/etc during initialization.
This is why HIVE-24157 introduced this  restriction as part of the in the resolver

Reply:
This means that for all the other kind of strict checks we are vulnerable to conversions happening during init right?

Reply:
Hey Stamati, it seems that if if its not a direct conversion that can be checked with LOSSY_CONVERSIONS, it has to be part of the UDF init. For example:

-  Bigint cannot be cast to a double: https://github.com/apache/hive/blob/adf426a87e60e0b99dec7e8aee7d1a5e12462161/ql/src/java/org/apache/hadoop/hive/ql/plan/ExprNodeGenericFuncDesc.java#L248
- Date only allows timestamps and Strings https://github.com/apache/hive/blob/853e1d1f9d17ef61637f094b7bd2cccf44931124/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFToDate.java#L68

Reply:
Small observation: the first example does not reflect the situation in current master. In HIVE-23100, the checks were moved from `ExprNodeGenericFuncDesc` to `TypeCheckProcFactory`. 

It is true that at the moment `UDFToBoolean`, `UDFToByte`, `UDFToDouble`, etc., are not checked by `TypeCheckProcFactory`; I am not sure if this needs to change or not. In that sense if we want to enforce type checks we need to do it in another way as it is done here with the resolver. 

The timestamp resolver only checks the case for timestamp/dates but there are other lossy conversions (e.g., BigInt to double) for which we are doing nothing. I assume that if somebody does `CAST( bigint as double)` the query will run without raising an error even if strict checks are enabled. 

Taking it on the other way around currently if somebody does `CAST (date as double)` the query will run or fail depending on the value of the strict checks property.

Reply:
Putting a note here so that we don't forget. The SQL standard forbids all kind of explicit casts from DATE, TIME, DATETIME to non char datatypes and does not mention anything about implicit casts.